src/dataset/create_dataset.py

```diff
@@ -19,240 +20,141 @@
-                        print(f'{entry["source"]}')
-                        continue
-                    
-                    #if the text is too short, ignore this text
-                    if len(text) < 500:
-                        continue
-
-                    #we're keeping the text so we inc the aticle count
-                    self.articles_count[entry['source']] += 1
-                    self.total_articles_count += 1
-                    
-                    # Get signature
-                    signature = ""
-                    if title: signature += f"Title: {title}, "
-                    else: signature += f"Title: None, "
-                    if author: signature += f"Author: {author}"
-                    else: signature += f"Author: None"
-                    # if date_published: signature += f"Date published: {date_published}, "
-                    # if url: signature += f"URL: {url}, "
-                    # if tags: signature += f"Tags: {tags}, "  # Temporary decision to not include tags in the signature
-                    # if signature: signature = signature[:-2]
-                    signature = signature.replace("\n", " ")
-                    
-                    # Add info to metadata and embedding strings
-                    self.metadata.append((title, author, date_published, url, tags))
-                    blocks = text_splitter.split(text, signature)
-                    self.embedding_strings.extend(blocks)
-                    self.embeddings_metadata_index.extend([self.total_articles_count-1] * len(blocks))
-                    
-                    # Update counts
-                    self.total_char_count += len(text)
-                    self.total_word_count += len(text.split())
-                    self.total_sentence_count += len(split_into_sentences(text))
-                    self.total_block_count += len(blocks)
-                    
-                except MissingDataException as e:
-                    if str(e) not in error_count_dict:
-                        error_count_dict[str(e)] = 0
-                    error_count_dict[str(e)] += 1
+
+        # Load the dataset. streaming allows you to load one entry at a time, 
+        # so entries can be processed before the entire dataset has been saved.
+        # iterable_data = load_dataset('StampyAI/alignment-research-dataset', 'aisafety.info', split='train', streaming=True)
+
+        iterable_data = load_dataset('StampyAI/alignment-research-dataset', 'all', split='train', streaming=True)
+        
+        for entry in tqdm(iterable_data):            
+            """Checks"""
+
+            # if we specified custom sources, only include entries from those sources
+            if (self.custom_sources is not None) and (entry['source'] not in self.custom_sources):
+                continue
+
+            # raise error if the entry does not contain the required metadata
+            self.contains_required_metadata(entry)
```

Comment:
this means that a single faulty item will blow up the whole run - is that desired?

Reply:
I think so, because at this point we want to make sure the ARD has correct entries. If it doesn't, it indicates some of its code has to be changed asap to be standardized. Although I can use logging.error instead? Like would do something like
```
# raise error if the entry does not contain the required metadata
try:
    self._validate_required_metadata(entry)
except MissingDataException as mde:
    logging.error(str(mde))  # Log the error message
except Exception as e:
    raise e
```
Although, I don't have a logging thing yet, will have to look at ARD to see how they did it